src/Settings.cpp

```diff
@@ -155,3 +177,42 @@ bool Settings::getStartInTray() const
 
     return enable;
 }
+
+void Settings::setAutostart(bool autostart)
+{
+    m_keyFile.set_boolean(GROUP_GENERAL, "autostart", autostart);
+}
+
+bool Settings::getAutostart()
+{
+    auto autostart = false;
+    try
+    {
+        autostart = m_keyFile.get_boolean(GROUP_GENERAL, "autostart");
+    }
+    catch (Glib::KeyFileError const& error)
+    {
+        std::cerr << "Settings: " << error.what() << std::endl;
+    }
+    return autostart;
+}
+
+void Settings::autostart()
+{
+    auto dfPath = getDesktopFilePath();
+    if(dfPath.empty())
+    {
+        std::cerr << "Settings: Failed to find desktop file" << std::endl;
+        return;
+    }
+
+    auto homeDir = std::getenv("HOME");
+    auto gfileDestPath = strcat(homeDir, "/.config/autostart/whatsapp-for-linux.desktop");
+    auto gfileSrc = Gio::File::create_for_path(dfPath);
+    auto gfileDest = Gio::File::create_for_path(gfileDestPath);
+
+    if(Settings::getInstance().getAutostart())
```

Comment:
Do the appropriate checks to see if the file already exists before copying to prevent unnecessary operation.

Reply:
I already do that here

```cpp
    if(dfPath.empty())
    {
        std::cerr << "Settings: Failed to find desktop file" << std::endl;
        return;
    }
 ```